frontend/src/hooks/useShortAddress.ts

```diff
@@ -0,0 +1,14 @@
+export const useShortAddress = () => {
+  function shortenAddress(address?: string) {
+    const prefix = address?.startsWith('0x') ? '0x' : '';
```

Comment:
Maybe this is a bit more readable, what do you think? 
```suggestion
    const prefix = '0x';
    const addressWithoutPrefix = address?.replace('0x', '')
    return `${prefix}${address.slice(0,4)}...${address.slice(-4)}
```

Reply:
actually already have a simplified version in upstream branch, will cherry pick and you can check